src/auth/index.ts

```
@@ -12,8 +12,8 @@ export class AuthManager {
     this.authServerUrl = config.authServerUrl;
   }
 
-  async verifyToken(token: string): Promise<boolean> {
-    if (!token) return false;
+  async verifyToken(token: string): Promise<{ userUid: string, [key: string]: any } | null> {
+    if (!token) return null;
```

How about throwing here instead of returning `null`?